dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ReferenceConfig.java

```diff
@@ -309,23 +273,20 @@ private void init() {
         appendParameters(map, module);
         appendParameters(map, consumer, Constants.DEFAULT_KEY);
         appendParameters(map, this);
-        String prefix = StringUtils.getServiceKey(map);
+        Map<String, Object> attributes = null;
         if (methods != null && !methods.isEmpty()) {
-            Map<Object, Object> attributes = new HashMap<Object, Object>();
-            for (MethodConfig method : methods) {
-                appendParameters(map, method, method.getName());
-                String retryKey = method.getName() + ".retry";
+            attributes = new HashMap<String, Object>();
+            for (MethodConfig methodConfig : methods) {
+                appendParameters(map, methodConfig, methodConfig.getName());
+                String retryKey = methodConfig.getName() + ".retry";
                 if (map.containsKey(retryKey)) {
                     String retryValue = map.remove(retryKey);
                     if ("false".equals(retryValue)) {
-                        map.put(method.getName() + ".retries", "0");
+                        map.put(methodConfig.getName() + ".retries", "0");
                     }
                 }
-                appendAttributes(attributes, method, prefix + "." + method.getName());
-                checkAndConvertImplicitConfig(method, map, attributes);
+                attributes.put(methodConfig.getName(), convertMethodConfig2AyncInfo(methodConfig));

```

Comment:
why remove `appendAttributes(attributes, method, prefix + "." + method.getName());`

Reply:
```
appendAttributes(attributes, method, prefix + "." + method.getName());
checkAndConvertImplicitConfig(method, map, attributes);
```
Those above methods is only for the storing async attributes to StaticContext.